src/Features/LanguageServer/Microsoft.CodeAnalysis.LanguageServer/HostWorkspace/LanguageServerWorkspace.cs

```diff
@@ -89,27 +103,33 @@ internal override ValueTask TryOnDocumentClosedAsync(DocumentId documentId, Canc
         return this.ProjectSystemProjectFactory.ApplyChangeToWorkspaceAsync(
             async w =>
             {
-                // TODO(cyrusn): This only works for normal documents currently.  We'll have to rethink how things work
-                // in the world if we ever support additionalfiles/editorconfig in our language server.
-                var document = w.CurrentSolution.GetDocument(documentId);
+                var textDocument = w.CurrentSolution.GetDocument(documentId) ?? w.CurrentSolution.GetAdditionalDocument(documentId);
 
-                if (document is { FilePath: { } filePath })
+                if (textDocument is { FilePath: { } filePath })
                 {
                     TextLoader loader;
-                    if (document.DocumentState.Attributes.DesignTimeOnly)
+                    var document = textDocument as Document;
+                    if (document?.DocumentState.Attributes.DesignTimeOnly == true)
                     {
                         // Dynamic files don't exist on disk so if we were to use the FileTextLoader we'd effectively be emptying out the document.
                         // We also assume they're not user editable, and hence can't have "unsaved" changes that are expected to go away on close.
                         // Instead we just maintain their current state as per the LSP view of the world.
-                        var documentText = await document.GetTextAsync(cancellationToken);
+                        var documentText = await textDocument.GetTextAsync(cancellationToken);
                         loader = new SourceTextLoader(documentText, filePath);
                     }
                     else
                     {
                         loader = this.ProjectSystemProjectFactory.CreateFileTextLoader(filePath);
                     }
 
-                    this.OnDocumentClosedEx(documentId, loader, requireDocumentPresentAndOpen: false);
+                    if (document is not null)
+                    {
+                        this.OnDocumentClosedEx(documentId, loader, requireDocumentPresentAndOpen: false);
+                    }
+                    else
+                    {
+                        this.OnAdditionalDocumentClosed(documentId, loader, requireDocumentPresentAndOpen: false);
```

Comment:
given that the underlying methods we're deferring to are distinguished by Doc vs AdditionalDoc i'm now like 88% of the way to thinking these calling methods shoudl be the same.